PSI_Tracker.js

```diff
@@ -0,0 +1,681 @@
+
+  SpreadsheetApp.flush();
+  // STEP 4: If there are more URLs to run, cycle through this function again
+  toast("Done", "Batch finished");
+  if (batch.length >= NUMBER_OF_URLS_PER_BATCH) {
+    return true;
+  }
+  // if not, we exit and finish the program
+  removeBatchTriggers();
+  return false;
+}
+
+/**
+ * To parse the results from the PSI API, we rely on the tab "Fields" that are going to contain the functions to run
+ */
+function parseResults(api, fetch, batch, sheet, save_screenshot = false) {
+  // STEP 1: We retrieve the fields we want to extract from the JSON from the tab "Fields"
+  // They are made of the following elements:
+  // 1. A "Method", which corresponds to the current API being run (here we will focus only on PSI API)
+  // 2. A "Field" which is the name of the field calculated and will match the "Results" tab
+  // 3. A "Data" which contains the way to access the information from the "content" field from the response
+  // Example: "LH Version" reads `content["lighthouseResult"]["lighthouseVersion"]` to read the Lighthouse version from the API
+  const sheet_fields = FIELD_SHEET;
+  const sheet_fields_last_row = sheet_fields.getLastRow();
+  let fields = sheet_fields
+    .getRange(1, 1, sheet_fields_last_row, 3)
+    .getValues();
+  fields = fields.map((d) => {
+    return {
+      method: d[0],
+      field: d[1],
+      data: d[2],
+    };
+  });
+  // We only focus on the fields matching the PSI API method
+  const dict_matching_api_names = {
+    PSI: ["PSI API"],
+    CrUX: ["CrUX"],
+    "CrUX History": ["CrUX History"],
+    "Green Domain": ["Green Domain"],
+    Accessibility: ["Accessibility"],
+    Sustainability: ["Sustainability"],
+    PSI_A11Y_S12Y: ["Accessibility", "PSI API", "Sustainability"],
+  };
+  // Extract the Fields matching any element in the nacthing dict
+  fields = fields.filter(
+    (d) => dict_matching_api_names[api].indexOf(d.method) !== -1
+  );
+  // Surface Green Domains list to be accessed by the fields extracted
+  let green_domains = [];
+  if (api === "Sustainability" || api === "PSI_A11Y_S12Y") {
+    green_domains = SpreadsheetApp.getActiveSpreadsheet()
+      .getSheetByName("Green Domains (GWF)")
+      .getRange("F2:J")
+      .getValues()
+      .filter((d) => d[4] === true)
+      .map((d) => d[0]);
+  }
+
+  // STEP 2: We iterate through all the fields
+  for (let i = 0; i < fetch.length; i++) {
+    const fields_for_debug = [
+      new Date().toISOString().slice(0, 10),
+      batch[i].label,
+      batch[i].url,
+      batch[i].device,
+      batch[i].origin,
+    ];
+    let debug = "";
+    // Indicate that we worked on that row in the Config tab
+    sheet.getRange(batch[i].id + 1, COLUMN_STATUS + 1).setValue("✅");
+    // Read and parse the data for that specific cakk
+    const data = fetch[i];
+    let content = {};
+    // Try to parse the answer from the API
+    try {
+      content = JSON.parse(data);
+    } catch (e) {
```

Comment:
Here and in the error case below, an early return (or here a continue) should be used and the rest of the function can be taken out of the else to reduce the complexity.